### PR TITLE
CI: Remove powershell scripts (Azure)

### LINKED ISSUE #26344: CI: Use bash for Windows scripts on Azure
Azure now supports using bash in windows scripts, so I _think_ we can de-duplicate some of the windows-specific shell scripts.  cc @datapythonista

### PR DESCRIPTION
**Pull request by alimcmaster1**
- [x] closes #26344

Ref @jbrockmendel comment here https://github.com/pandas-dev/pandas/pull/23454#discussion_r360933924

CI: fails as expected example [here](https://dev.azure.com/pandas-dev/pandas/_build/results?buildId=23715&view=logs&j=fd89438c-1fd3-5981-222e-115d06c89c2d)

### CODE REVIEW

File: pandas/tests/arrays/test_array.py [hunk 1]
```diff
@@ -343,7 +343,7 @@ def test_searchsorted(self, string_dtype):
 
         result = arr.searchsorted("a", side="left")
         assert is_scalar(result)
-        assert result == 0
+        assert result == 1
```
Comment (alimcmaster1) [inline pandas/tests/arrays/test_array.py]: Checking that test failures fail CI

File: ci/azure/windows.yml [hunk 2]
```diff
@@ -23,32 +23,33 @@ jobs:
...
         Write-Host "##vso[task.prependpath]$HOME/miniconda3/bin"
       displayName: 'Add conda to PATH'
+
     - script: conda update -q -n base conda
       displayName: 'Update conda'
+
     - bash: |
         conda env create -q --file ci\\deps\\azure-windows-$(CONDA_PY).yaml
       displayName: 'Create anaconda environment'
+
```
Comment (alimcmaster1) [inline ci/azure/windows.yml]: Making newline between each step consistent with `posix.yml`

**Comment by jbrockmendel** [comment]
Nice! Thanks for taking a look at this.  LGTM pending green.

File: ci/azure/posix.yml [hunk 3]
```diff
@@ -69,20 +69,13 @@ jobs:
       displayName: 'Build versions'
 
     - task: PublishTestResults@2
+      condition: succeededOrFailed()
```
Comment (WillAyd) [inline ci/azure/posix.yml]: Is there a particular reason to specify this instead of keeping as always?

**Review by WillAyd — COMMENTED** [review]
lgtm one question

File: ci/azure/posix.yml (hunk 3 above)
Comment (alimcmaster1) [reply ci/azure/posix.yml]: My understand was that `suceeded()` is the default condition on Azure as per [docs](https://docs.microsoft.com/en-us/azure/devops/pipelines/process/conditions?view=azure-devops&tabs=yaml):

```
By default, steps, jobs, and stages run if all previous steps/jobs have succeeded. It's as if you specified "condition: succeeded()" (see Job status functions).
```

And I thought we want to publish test results regardless of whether failures are produced? Or am I missing something?

Comment (WillAyd) [reply ci/azure/posix.yml]: My mistake. I guess ultimately my question is this versus `always` if it matters

Comment (alimcmaster1) [reply ci/azure/posix.yml]: Thanks for taking a look! Apparently `always()` will run even if the pipeline is cancelled -> that's the difference

File: ci/azure/windows.yml [hunk 4]
```diff
@@ -23,32 +23,33 @@ jobs:
...
     - bash: |
         source activate pandas-dev
         conda list
         ci\\incremental\\build.cmd
       displayName: 'Build'
+
```
Comment (jbrockmendel) [inline ci/azure/windows.yml]: #30393 should have gotten rid of the build.cmd on L37.  this isnt going to re-introduce it by accident?

Comment (alimcmaster1) [reply ci/azure/windows.yml]: Apologies just merged master

**Comment by jreback** [comment]
is the net effect of this that azure will run all steps but fail the build if anyone of them fails?

**Comment by alimcmaster1** [comment]
> is the net effect of this that azure will run all steps but fail the build if anyone of them fails?

Goal behind this is https://github.com/pandas-dev/pandas/issues/26344 to get rid of the powershell scripts. (Azure step conditions are unchanged - apart from Publish test result - which will run regardless of the outcome of previous)

**Comment by jreback** [comment]
cool, thanks @alimcmaster1

**Pull request merged**